src/argilla/server/contexts/datasets.py

```diff
@@ -189,6 +189,17 @@ def get_record_by_id(db: Session, record_id: UUID):
     return db.get(Record, record_id)
 
 
+def get_records_by_ids(
+    db: Session, dataset_id: UUID, record_ids: List[UUID], include: List[RecordInclude] = []
+) -> List[Record]:
+    query = db.query(Record).filter(Record.dataset_id == dataset_id, Record.id.in_(record_ids))
+    if RecordInclude.responses in include:
+        query = query.options(joinedload(Record.responses))
```

Comment:
We should return only connected user responses (if any) instead of all responses (this will be enabled in a separate endpoint for admins). Take a look [here](https://calendar.google.com/calendar/u/0/r?cid=c_09f749125621f991d957be9dd672970956530dc9e4e2c647a9bdfa7c35a78ba7@group.calendar.google.com&pli=1)

Reply:
True! I will update this and add some unit tests.